homeassistant/components/heatzy/climate.py

```diff
@@ -0,0 +1,295 @@
+from .const import (
+    CFT_TEMP_H,
+    CFT_TEMP_L,
+    CONF_ALIAS,
+    CONF_ATTR,
+    CONF_ATTRS,
+    CONF_MODE,
+    CONF_MODEL,
+    CONF_ON_OFF,
+    CONF_PRODUCT_KEY,
+    CONF_VERSION,
+    CUR_TEMP_H,
+    CUR_TEMP_L,
+    DOMAIN,
+    ECO_TEMP_H,
+    ECO_TEMP_L,
+    ELEC_PRO_SOC,
+    GLOW,
+    PILOTEV1,
+    PILOTEV2,
+)
+
+MODE_LIST = [HVACMode.HEAT, HVACMode.OFF]
+PRESET_LIST = [PRESET_NONE, PRESET_COMFORT, PRESET_ECO, PRESET_AWAY]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Load all Heatzy devices."""
+    coordinator = hass.data[DOMAIN][entry.entry_id]
+    entities: list[HeatzyThermostat] = []
+    for unique_id, device in coordinator.data.items():
+        product_key = device.get(CONF_PRODUCT_KEY)
+        if product_key in PILOTEV1:
+            entities.append(HeatzyPiloteV1Thermostat(coordinator, unique_id))
+        elif product_key in PILOTEV2 or product_key in ELEC_PRO_SOC:
+            entities.append(HeatzyPiloteV2Thermostat(coordinator, unique_id))
+        elif product_key in GLOW:
+            entities.append(Glowv1Thermostat(coordinator, unique_id))
+    async_add_entities(entities)
+
+
+class HeatzyThermostat(CoordinatorEntity[HeatzyDataUpdateCoordinator], ClimateEntity):
+    """Heatzy climate."""
+
+    _attr_hvac_modes = MODE_LIST
+    _attr_preset_modes = PRESET_LIST
+    _attr_supported_features = ClimateEntityFeature.PRESET_MODE
+    _attr_temperature_unit = TEMP_CELSIUS
+    _attr_has_entity_name = True
+
+    def __init__(
+        self, coordinator: HeatzyDataUpdateCoordinator, unique_id: str
+    ) -> None:
+        """Init."""
+        super().__init__(coordinator)
+        self._attr_unique_id = unique_id
+        self._attr_name = coordinator.data[unique_id][CONF_ALIAS]
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, unique_id)},
+            name=coordinator.data[unique_id][CONF_ALIAS],
+            manufacturer=DOMAIN,
+            sw_version=coordinator.data[unique_id].get(CONF_VERSION),
+            model=coordinator.data[unique_id].get(CONF_MODEL),
+        )
+
+    @property
+    def hvac_mode(self) -> HVACMode:
+        """Return hvac operation ie. heat, cool mode."""
+        if self.preset_mode == PRESET_NONE:
+            return HVACMode.OFF
+        return HVACMode.HEAT
+
+    async def async_set_hvac_mode(self, hvac_mode: HVACMode) -> None:
+        """Set new hvac mode."""
+        if hvac_mode == HVACMode.OFF:
+            await self.async_turn_off()
+        elif hvac_mode == HVACMode.HEAT:
+            await self.async_turn_on()
+
+    async def async_turn_on(self) -> None:
+        """Turn device on."""
+        await self.async_set_preset_mode(PRESET_COMFORT)
+
+    async def async_turn_off(self) -> None:
+        """Turn device off."""
+        await self.async_set_preset_mode(PRESET_NONE)
+
```

Comment:
The idiom `self.coordinator.data[self.unique_id].get(CONF_ATTR, {})` repeats so many times in all of the subclasses that it would be good to lift into a property (naming could be different) here:

```suggestion
    @property
    def _conf_attr(self):
        return self.coordinator.data[self.unique_id].get(CONF_ATTR, {})
```

Reply:
it's actually cleaner. It is done.